tools/ports/__init__.py

```diff
@@ -343,16 +384,17 @@ def up_to_date():
 
     # before acquiring the lock we have an early out if the port already exists
     if up_to_date():
-      return
+      return False
 
     # main logic. do this under a cache lock, since we don't want multiple jobs to
     # retrieve the same port at once
+    cache.ensure()
```

Comment:
This seems a little arbitrary to add this here.  Hopefully we can find a better place for it.

Reply:
That is where I was blocked and that is how I was able to move forward. If you can tell me where to put that logic which would make more sense I will gladly do it.